test/test_network.h

```diff
@@ -156,26 +156,23 @@ TEST(network, multi_out) {
     auto hidden = std::make_shared<layers::fc<tan_h>>(2,4);
     auto out1 = std::make_shared<layers::fc<tan_h>>(4,2);
     auto out2 = std::make_shared<layers::fc<tan_h>>(4,2);
-    auto out3 = std::make_shared<layers::fc<tan_h>>(4,2);
 
     in << hidden << out1;
     hidden << out2;
-    hidden << out3;
 
     for (size_t i = 0; i < tnum; i++) {
         bool in[2] = { bernoulli(0.5), bernoulli(0.5) };
         label_t expected = (in[0] ^ in[1]) ? 1 : 0;
         data.push_back({ in[0] * 1.0, in[1] * 1.0 });
 
         out.emplace_back(std::vector<vec_t>{
-            {(in[0]&&in[1])*1.0, (in[0]||in[1])*1.0}, // 1st output: train and/or function
-            {(in[0]^in[1])*1.0, (in[0]==in[1])*1.0},  // 2nd output: train xor/eq function
-            {-1, 0.5}                                 // 3rd output: train constants (purpose: make output channel count different from output dimension)
+            {(in[0]&&in[1])*1.0, (in[0]||in[1])*1.0}, // 1st output train and/or function
+            {(in[0]^in[1])*1.0, (in[0]==in[1])*1.0} // 2nd output train xor/eq function
```

Comment:
@reunanen please cast the results to `float_t` as done [here](https://github.com/edgarriba/tiny-cnn/blob/8756f3f8a9a98420ae5b69c96d7394866f6c3d6e/test/test_network.h#L166) in order to have a consistent code